platformio.ini

```diff
@@ -0,0 +1,151 @@
+# This is the configuration file for PlatformIO, a cross-platform tool and IDE
+# to build and develop software for embedded systems.
+#
+# To build the firmware with PlatformIO, install PlatformIO Core (CLI, no GUI):
+# - https://docs.platformio.org/en/latest//core/installation.html
+#
+# For documentation on this file, see:
+# - https://docs.platformio.org/en/latest/projectconf/index.html
+#
+# To add support for another STM32F1xx board, add a new [env:name..] section
+# with the appropriate `board` and (optionally) `board_build.ldscript`.
+# For a list of supported hardware, see:
+# - https://docs.platformio.org/en/latest/platforms/ststm32.html
+#
+# To add support for additional grblHAL plugins or drivers, make the necessary
+# changes to the common.build_flags, common.lib_deps and common.lib_extra_dirs
+# below.
+#
+# By default this file sets the preprocessor directive `OVERRIDE_MY_MACHINE`
+# to ignore all settings in `Inc/my_machine.h`.  Instead, you should edit this
+# file and add the necessary directives (e.g. `-D FOOBAR_ENABLE`) to either
+# `common.build_flags` or the environment specific `build_flags`.
+#
+# Typical command line usage:
+#
+#    # Build firmware for all defined environments
+#    pio run
+#    # Build only for specific environments with the -e option
+#    pio run -e BTT_SKR_MINI_E3_V20 -e BTT_SKR_MINI_E3_V20_USB
+#    # List the freshly built firmware ELF (.elf) and binary (.bin) files:
+#    ls -l .pio/build/*/firmware.*
+#    # If the target supports uploading (via `upload_protocol`):
+#    pio run -e bluepill_f103c8_128k -t upload
+#    # Clean up build related resources
+#    pio run -t clean
+#
+[platformio]
+include_dir = Inc
+src_dir = Src
+
+[common]
+build_flags =
+  -I .
+  -I FatFS
+  -I Middlewares/ST/STM32_USB_Device_Library/Class/CDC/Inc
+  -I Middlewares/ST/STM32_USB_Device_Library/Core/Inc
+  -I USB_DEVICE/Target
+  -D RX_BUFFER_SIZE=512
```

Comment:
These settings are for limiting RAM use on the xB (128K flash) variant, there is plenty of RAM available on the RC so can be left out for that (default values are then used). Not very important for now but worth mentioning.

I have added settings descriptions as an option, currently all drivers except for the F103 xB variant has enough flash for that. To allow grblHAL to compile for that variant the symbol NO_SETTINGS_DESCRIPTIONS has to be added. I have not checked if it can be left out for a non-USB build (I guess so), but then most will want to build with USB enabled?

Reply:
> In the future, automated build tests may be implemented using GitHub Actions

I know - but I think it is a bit early to add that now. Automated build is set for the ESP32 driver (by a PR) and it fails - it has to be debugged. And which build options to run automated builds has to be determined?

Should I merge this PR as it is now? I can add the NO_SETTINGS_DESCRIPTIONS in the commit where it is required.

Reply:
Thanks for the feedback. I'll get back to you in a few days when I have time to work on this again.

Reply:
I have added your platformio.ini to the repo manually, thanks for that.

Reply:
That sounds good, thanks for the feedback.  And sorry for letting this hang around for so long.. life got in the way.